provider/docker_test.go

```diff
@@ -2271,3 +2271,613 @@ func TestSwarmTaskParsing(t *testing.T) {
 		}
 	}
 }
+
+func TestDockerGetServiceProtocol(t *testing.T) {
```

Comment:
I'll have to work on these, because our unit tests are way too verbose (but it's orthogonal to this PR, just thinking outloud 🤔 ).